### PR TITLE
[MM-20594] Migrate components/more_direct_channels to TypeScript

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Migration of `components/more_direct_channels` to TypeScript.
Done internally for use by the new channel sidebar.
Tests will be migrated at a later date to save time.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/13479
https://mattermost.atlassian.net/browse/MM-20594

### CODE REVIEW

**Comment by lindy65** [comment]
Thanks @devinbinnie - I'll QA after dev reviews are done...

**Comment by wiggin77** [comment]
@devinbinnie  I don't think my Typescript is strong enough to review this reliably.

File: components/more_direct_channels/index.ts [hunk 1]
```diff
@@ -22,19 +22,36 @@ import {
...
+    views: {
+        search: {
+            modalSearch: string;
+        };
+    };
+}
+
+type OwnProps = {
+    isExistingChannel: boolean;
+
```
Comment (hmhealey) [inline components/more_direct_channels/index.ts]: Blank line here

**Comment by lindy65** [comment]
Hi @mkraft - please ping me here once you've reviewed so I can QA :) Thanks!

**Comment by devinbinnie** [comment]
@lindy65 Ready for QA review

**Comment by lindy65** [comment]
Hey @devinbinnie 

Smoke-tested around DM and GM messages and I'm seeing the issue in https://mattermost.atlassian.net/browse/MM-22740 (which repro's for me with any DM and GM as well). 

Also, I see you mentioned above that "Tests will be migrated at a later date to save time" - Please link the Jira ticket here for tracking purposes... Thanks!

**Comment by devinbinnie** [comment]
> Hey @devinbinnie
> 
> Smoke-tested around DM and GM messages and I'm seeing the issue in https://mattermost.atlassian.net/browse/MM-22740 (which repro's for me with any DM and GM as well).
> 
> Also, I see you mentioned above that "Tests will be migrated at a later date to save time" - Please link the Jira ticket here for tracking purposes... Thanks!

Thanks @lindy65, here's a ticket for the tests: https://mattermost.atlassian.net/browse/MM-22768

Since the issue above isn't caused by this PR, can we go ahead and approve and merge this? Thanks again :)

**Pull request merged**